### PR TITLE
[backend/rust] use tracing crate and assign request ids to spans

### PR DESCRIPTION
**Pull request by problame**
* New request IDs are assigned to each request as soon as we get it from hyper (in `handle_request`)
* Their purpose is to enable attribution of log messages to a particular request
* Currently request IDs are only embedded into the root `tracing::Span` of each request
* [ ] should we include them in response headers?
  use case: correlate an unexpected response with log messages

### CODE REVIEW

File: humblegen-rt/src/server.rs [hunk 1]
```diff
@@ -52,8 +54,14 @@ pub async fn handle_request(
     services: Arc<RegexSetMap<Request<Body>, Service>>,
     req: Request<Body>,
 ) -> Response<Body> {
+    let request_id: String = rand::thread_rng()
+        .sample_iter(&rand::distributions::Alphanumeric)
+        .take(10)
```
Comment (mbr) [inline humblegen-rt/src/server.rs:59]: This is ~ 60 bits of entropy, correct? A uuid typically a little under 128, maybe we should go up to at least 20  chars here?

Comment (mbr) [reply humblegen-rt/src/server.rs:59]: According to http://davidjohnstone.net/pages/hash-collision-probability, with 60 bits and 10 million requests, we're already at 0.004% collision chance, which seems a bit too close for comfort... =)

Comment (problame) [reply humblegen-rt/src/server.rs:59]: Have you thought about the goal we are trying to achieve here? Attributing log entries, not having an almost-guaranteed collision-freeness.

**Comment by problame** [comment]
I'll give this another pass and add the response headers as well

**Pull request merged**